### PR TITLE
Remove driver version check 

### PR DESCRIPTION
**Pull request by zeeke**
Avoid checking the driver version against a user
defined map. If the netlink collector is not supported by
the Physical Function, the next provided collector (e.g. `sysfs`)
is used.

superseeds:
- https://github.com/k8snetworkplumbingwg/sriov-network-metrics-exporter/pull/35

cc @Eoghan1232 , @SchSeba

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,7 +9,10 @@ The SR-IOV Network Metrics Exporter is designed with the Kubernetes SR-IOV stack
 The sysfs collector for Virtual Function telemetry supports NICs with drivers that implement the SR-IOV sysfs management interface e.g. ice, i40e, mlnx_en and mlnx_ofed.
 
 The netlink collector relies on driver support and a kernel version of 4.4 or higher.
-To support netlink, we recommend these driver versions: an i40e driver of 2.11+ or higher for Intel® 700 series NICs and ice driver 1.2+ for Intel® 800 series NICs.
+To support netlink, we recommend these driver versions:
```
Comment (SchSeba) [inline README.md:12]: maybe something about minimal supported driver version or something like that?

**Review by SchSeba — COMMENTED** [review]
just a small nit and we can merge this one

File: README.md (hunk 1 above)
Comment (zeeke) [reply README.md]: I rephrased as:
```
The netlink collector relies on driver support and a kernel version of 4.4 or higher.
Also, the following minimum driver versions are required for this collector:
- `i40e` - 2.11+ for Intel® 700 series NICs 
- `ice` - 1.2+ for Intel® 800 series NICs
- `mlx5_core` - 5.15+ for Mellanox NICs
```

does it sound better?

**Comment by zeeke** [comment]
@Eoghan1232 please take a look at this

**Review by Eoghan1232 — APPROVED** [review]
I very much like this approach! 
Makes changes down the line easier too, without the hassle of driver validation due to so many variations of namings.
LGTM!

**Review by SchSeba — APPROVED** [review]
LGTM

nice work!

**Pull request merged**